### ISSUE TITLE
Incorrect newtype suggestion for existential types

### ISSUE THREAD

**Issue by Topsii** [ndmitchell/hlint#524]
```
{-# LANGUAGE ExistentialQuantification #-}

data Foo a = Eq a => MkFoo a
```
Gives
```
HLintTest.hs:3:1: Suggestion: Use newtype instead of data
Found:
  data Foo a = Eq a => MkFoo a
Perhaps:
  newtype Foo a = Eq a => MkFoo a
Note: decreases laziness
```
but this will fail to compile:
```
HLintTest.hs:3:17: error:
    * A newtype constructor cannot have a context in its type
      MkFoo :: forall a. Eq a => a -> Foo a
    * In the definition of data constructor `MkFoo'
      In the newtype declaration for `Foo'
  |
3 | newtype Foo a = Eq a => MkFoo a
  |                 ^^^^^^^^^^^^^^^
```

**Comment by ndmitchell** [comment]
Doh! Thanks for the report. Just pushed a fix and I'll make a release in the next week or so (unless it's causing you more problems, in which case I'll do so tonight).

**Comment by zoranbosnjak** [comment]
@ndmitchell the original problem as described above is resolved, but the same problem is still present with this scenario:

```haskell
{-# LANGUAGE ExistentialQuantification #-}                                                                 
{-# LANGUAGE GADTs #-}                                                                                     
                                                                                                           
data Some f where Some :: f a -> Some f                                                                    
data Foo t = Foo                                                                                           
data Bar = Bar (Some Foo) 
```

`hlint` suggests to use `newtype Bar...`, but this does not compile. Please consider re-opening the issue.

```
$ hlint --version
HLint v3.6.1, (C) Neil Mitchell 2006-2023
```
